VEDA_OS01/transcripts.py

```diff
@@ -790,133 +790,200 @@ def three_play_transcription_callback(sender, **kwargs):
+    except Exception:
+        # in case of any exception, log and raise.
+        LOGGER.exception(
+            u'[3PlayMedia Task] translation failed for video=%s -- lang_code=%s -- process_id=%s',
+            edx_video_id,
+            file_id,
+            target_language,
         )
+        raise
 
-        if not translation_status:
-            # Fail the process
-            translation_process.update(status=TranscriptStatus.FAILED)
-            continue
+    return sjson_file
 
-        # On a complete translation
-        if translation_status['state'] == COMPLETE:
 
-            # 1 - Fetch translation content from 3Play Media.
-            try:
-                srt_transcript = fetch_srt_data(
-                    url=THREE_PLAY_TRANSLATION_DOWNLOAD_URL.format(
-                        file_id=translation_process.process_id, translation_id=translation_process.translation_id
-                    ),
-                    apikey=three_play_secrets.api_key,
-                )
-            except TranscriptFetchError:
-                LOGGER.exception(
-                    u'[3PlayMedia Task] Translation download failed for video=%s -- lang_code=%s -- process_id=%s.',
-                    *log_args
-                )
+def handle_video_translations(video, translations, file_id, api_key, log_prefix):
+    """
+    It is a sub-module of `retrieve_three_play_translations` to handle
+    all the completed translations for a single video.
+
+    Arguments:
+        video: Video data object whose translations need to be handled here.
+        translations: A list containing translations metadata information received from 3play Media.
+        file_id: It is file identifier that is assigned to a Video by 3Play Media.
+        api_key: An api key to communicate to the 3Play Media.
+        log_prefix: A logging prefix used by the main process.
+
+    Steps include:
+        - Fetch translated transcript content from 3Play Media.
+        - Validate the content of received translated transcript.
+        - Convert translated SRT transcript to SJson format and upload it to S3.
+        - Update edx-val for a completed transcript.
+        - update transcript status for video in edx-val as well as edx-video-pipeline.
+    """
+    video_translation_processes = get_in_progress_translation_processes(video)
+    for translation_metadata in translations:
+
+        translation_id = translation_metadata['id']
+        translation_state = translation_metadata['state']
+        target_language = translation_metadata['target_language_iso_639_1_code']
+
+        if translation_state == COMPLETE:
+            # Fetch the corresponding tracking process.
+            translation_process = get_in_progress_translation_process(
+                video_translation_processes,
+                file_id=file_id,
+                translation_id=translation_id,
+                target_language=target_language
+            )
```

Comment:
I would replace this call & call at [line 952](https://github.com/edx/edx-video-pipeline/pull/44/files#diff-f82b0480568faf247efc91479533b6ccR952) with single method call which will return single process, since we don't need all processes.

Reply:
we need them in case where we need to fail them all.